### PR TITLE
chore: split out ui store part 1 [DET-8218]

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

- [ ] two semi-related todos in the code I rather leave out of this pr

next step: add an option to fetch UI state without importing the whole store but leaving ui accessible from the main useStore.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/shared/contexts/UIStore.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  chromeCollapsed: false,
+  darkLight: DarkLight.Light,
+  isPageHidden: false,
+  mode: Mode.System,
+  showChrome: true,
+  showSpinner: false,
+  theme: {} as Theme,
+};
+
+export enum StoreActionUI {
```
Comment (hamidzr) [inline webui/react/src/shared/contexts/UIStore.ts:23]: this is consistent with how we have these in Store but maybe we want a shorter name here

**Comment by hamidzr** [comment]
this should be failing intg-downstream test. I suspect it's because we're adding files and the way we're looking for changes skip those

```
MERGE_BASE="$(git merge-base upstream/master HEAD)"
DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE":webui/react/src HEAD:webui/react/src)"
if [ "$DIFF_DIRS" = "shared" ] ; then
  echo "shared code changed, running downstream test"
else
  echo "shared code not affected, halting job now"
  circleci-agent step halt
fi
```

https://determinedai.atlassian.net/browse/DET-8034

cc @determined-ai/infrastructure

**Review by keita-determined — APPROVED** [review]
LGTM

File: webui/react/src/contexts/Store.tsx [hunk 2]
```diff
@@ -48,7 +34,9 @@ export interface State {
   pinnedWorkspaces: Workspace[];
   pool: PoolOverview;
   resourcePools: ResourcePool[];
-  ui: UI;
+  ui: StateUI & {
+    omnibar: OmnibarState; // TODO: this should probably be out of ui
```
Comment (keita-determined) [inline webui/react/src/contexts/Store.tsx:38]: Does that mean omnibar will be completely removed?

Comment (hamidzr) [reply webui/react/src/contexts/Store.tsx:38]: by ui I mean not nested under ui. I'm not sure if it's the correct move or not yet

Comment (keita-determined) [reply webui/react/src/contexts/Store.tsx:38]: Hmm if its flexible and easy to change in the future, either way is fine?

**Pull request merged**